nodeup/pkg/model/convenience.go

```diff
@@ -27,3 +34,31 @@ func s(v string) *string {
 func i64(v int64) *int64 {
 	return fi.Int64(v)
 }
+
+func getProxyEnvVars(proxies *kops.EgressProxySpec) []v1.EnvVar {
+	glog.Info("in getProxyEnvVars")
+	if proxies == nil {
+		glog.Info("proxies is == nil, returning empty list")
+		return []v1.EnvVar{}
+	}
+
+	if proxies.HTTPProxy.Host == "" {
+		glog.Warning("EgressProxy set but no proxy host provided")
```

Comment:
returning an error just forces the caller to deal with the situation right?  I'm not 100% sure either way.  Things are probably going to fail to come up in this situation, we defiantly want to catch this in validation in any case, but as to whether we should return an error here, since I'm not sure, I'm just going to lean towards YAGNI and leave it as is.